gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/store/TypeStore.java

```diff
@@ -33,18 +36,42 @@ public TypeStore(List<Class<?>> concreteClasses) {
   }
 
   private void putConcreteClassses(List<Class<?>> concreteClasses) {
-    store.putAll(
-        concreteClasses.stream()
-            .collect(
-                Collectors.toMap(
-                    Class::getSimpleName,
-                    c -> TypeNode.withReference(ConcreteReference.withClazz(c)))));
+    for (Class<?> clazz : concreteClasses) {
+      store.put(clazz.getSimpleName(), TypeNode.withReference(ConcreteReference.withClazz(clazz)));
+      clazzStore.put(clazz.getSimpleName(), clazz);
+    }
   }
 
   public TypeNode get(String typeName) {
     return store.get(typeName);
   }
 
+  public TypeNode getWithFullName(String typeName) {
+    TypeNode stored = store.get(typeName);
+    Preconditions.checkNotNull(stored);
+    Reference ref;
+    if (stored.reference() instanceof ConcreteReference) {
+      ref =
```

Comment:
I'm afraid we might be losing information here, I see that `ConcreteReference` has a lot more fields other than `clazz` and `useFullName`, same thing for `VaporReference`.

Reply:
I moved the logic to `createImplementsTypes` by using each one of the setters in the builder